lib/wasix/src/runtime/resolver/wapm_source.rs

```diff
@@ -144,6 +208,149 @@ fn decode_summary(pkg_version: WapmWebQueryGetPackageVersion) -> Result<PackageS
     })
 }
 
+/// A local cache for package queries.
+#[derive(Debug, Clone)]
+struct FileSystemCache {
+    cache_dir: PathBuf,
+    timeout: Duration,
+}
+
+impl FileSystemCache {
+    const DEFAULT_TIMEOUT: Duration = Duration::from_secs(10 * 60);
+
+    fn new(cache_dir: impl Into<PathBuf>) -> Self {
```

Comment:
Can we make the timeout configurable?

I'd might need to play around with it on Deploy.

Reply:
I've added a `timeout: Duration` parameter to the `WapmSource::with_local_cache()` method :+1: